### PR TITLE
sorry for the chaotic commits - problems with the GitKraken client

### PR DESCRIPTION
**Pull request by daszunia**
Finally managed to separate branches. Sorry for the inconvenience.

### CODE REVIEW

**Comment by daszunia** [comment]
I'm working on the documentation file now

File: Sources/Administrator/InputHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#include <iostream>
+#include <cstring>
+
+void InputHandler::run()
+{
+    std::cout << "Waiting for a command:\n";
+    std::string command;
+    std::string name;
+
+    while ( name != "exit")
```
Comment (Kacper20) [inline Sources/Administrator/InputHandler.cpp]: Maybe do while loop would be better? Just suggestion, but i think in this case code is better structured in `do {} while`

File: Sources/Administrator/InputHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    {
+        std::getline (std::cin,command);
+
+        if ( command.find(" ") == std::string::npos )
+            name = command;
+        else name = command.substr(0,command.find_first_of(" "));
```
Comment (Kacper20) [inline Sources/Administrator/InputHandler.cpp]: code style: I would keep braces - this ifs with one line are sneaky. Or if not - be consistent and move code after else to line below. 
We could also write it like that: 
`auto name = command.find(" ") == std::string::npos ? command : command.substr(0,command.find_first_of(" "))`

File: Sources/Administrator/InputHandler.cpp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+        if ( name == "help") {
+        printHelp();
```
Comment (Kacper20) [inline Sources/Administrator/InputHandler.cpp]: Let's keep indentation consistent across the project - 2 spaces, google style code in Clion.

File: Sources/Administrator/InputHandler.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        }
+        else if (  name == "launch_process" ) {
+        launchProcess(command);
+        }
+        else {
+        std::cout << "Incorrect command.\n";
+        printHelp();
+        }
+    }
+    std::cout << "Exiting Administrator process. Goodbye.\n";
```
Comment (Kacper20) [inline Sources/Administrator/InputHandler.cpp]: We should exit this process here.

File: Sources/Administrator/InputHandler.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+public:
+    void run();
+    void printHelp();
+    int checkAddress(std::string address_with_port);
+    void handleConnect(std::string full_command);
+    void handleDisconnect();
+    void sendProcess(std::string full_command);
+    void launchProcess(std::string full_command);
+};
```
Comment (Kacper20) [inline Sources/Administrator/InputHandler.h:22]: I believe all of this except run should be private.

File: Sources/Administrator/InputHandler.cpp [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+
+void InputHandler::printHelp()
+{
```
Comment (Kacper20) [inline Sources/Administrator/InputHandler.cpp:53]: maybe using namespace std in all of the file? Readability would improve 👍

File: Sources/Administrator/InputHandler.h [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+{
+
+public:
+    void run();
+    void printHelp();
+    int checkAddress(std::string address_with_port);
+    void handleConnect(std::string full_command);
+    void handleDisconnect();
+    void sendProcess(std::string full_command);
+    void launchProcess(std::string full_command);
```
Comment (Kacper20) [inline Sources/Administrator/InputHandler.h]: `const std::string& command`

**Comment by Kacper20** [comment]
Good job. :clap Take a look at few comments

File: Sources/Administrator/InputHandler.cpp (hunk 6 above)
Comment (daszunia) [inline Sources/Administrator/InputHandler.cpp:53]: I checked other files and there wasn't namespace used so I thought that we shouldn't use it :/

**Comment by Kacper20** [comment]
Thats an open thing - in case like this whem we have almost only printing things to console it could be good. But nothing happens if it stays the same. :-) if you want to change it fast - find replace on std:: replaced by empty string.

File: Documentation/Administrator_functions.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+### Available functions for administrator
+
+#### Connect
+_connect <"server address:port">_
+Connects to server on the specified address (address provided with port). 
+
+#### Sending process
+_send_process <"name"> <"path to file">_
+User specifies the <"path to the file"> which will be sent to the server. The file should be on the user's machine.
+The argument <"name"> is the process's name on the server.
```
Comment (Kacper20) [inline Documentation/Administrator_functions.md:12]: That's nice.
Only one note to detail: We need to remember that only admin has access to name of process given by user. So in admin we should have map<string, string>, where key is a name g. You're creating unique UUID, that is inserted as value for given key(we have functions for creating this in C++ for sure). So only UUID is send to the server(and later to node). Later, you're receiving UUID, and doing mapping to name.

So maybe not a map, but rather set<tuple<string, string>>. :) To support fast find in every case. 
So basically user could write "processss1", but node and server sees only unique identifier given by node.

**Comment by Kacper20** [comment]
Good job. Please read one comment - this is irrelevant here, but needed for later implementation.
👏
